mautic/mautic
# [UI/UX] Campaign builder events

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 



## Description
This PR is based [on another PR](https://github.com/mautic/mautic/pull/13735), where I introduce tokens to replace most of the UI variables, aiming to improve consistency in all components. **We need it to be merged before this one becomes available for code review.**

Other dependencies:
- #13772

It enhances UX and UI for the builder.

Specifically:
- Improve system feedback while building campaigns, making easier to know what kind of event is that one
- Allows users to use the Name field on each added block to provide more context, so they can organize campaigns logically
- Increases the visual difference between decisions, actions and conditions
- Adds icons for each type of decision, action or condition
- Pushes Mautic closer to commercial solutions in the market (eg. Hubspot etc)
- Adds support for dark theme
- Creates a new style 
- Adds a _(really)_ subtle grid (which we can change opacity if you find too transparent)
- Updates button icons
- Fixes smaller UI issues
- Improves the experience for the Jump action block




| Before                                 | After
| -------------------------------------- | ---
|  ![image](https://github.com/mautic/mautic/assets/116097999/ef33809e-e616-4678-82a0-3e65a45ce2d1)                                       | ![image](https://github.com/mautic/mautic/assets/116097999/c1be0aad-0359-4321-a93e-65b46403b0a8)

---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Campaigns
3. Create one
4. Fill the name
5. Open the builder
6. Add blocks and see
